AdyenCard/Components/Card/CardComponent.swift

```diff
@@ -282,12 +280,22 @@ public final class CardComponent: PaymentComponent, PresentableComponent, Locali
         return footerItem
     }()
     
-    fileprivate func requestCardType(for bin: String, update item: FormCardNumberItem) {
+    private func didReceived(bin: String) {
+        let quickDetection = supportedCardTypes.adyen.types(forCardNumber: bin)
+        self.securityCodeItem.selectedCard = quickDetection.first
+        self.numberItem.numberFormatDidChange(detectedCards: quickDetection)
+        self.cardComponentDelegate?.didChangeBIN(String(bin.prefix(CardComponent.publicBinLenght)), component: self)
+        
+        throttler.throttle { [weak self] in
+            self?.requestCardTypes(for: bin)
+        }
+    }
+    
+    private func requestCardTypes(for bin: String) {
         cardTypeProvider.requestCardType(for: bin, supported: self.supportedCardTypes) { [weak self] cardTypes in
             guard let self = self else { return }
             
-            self.securityCodeItem.selectedCard = cardTypes.first
-            item.detectedCardsDidChange(detectedCards: cardTypes)
+            self.numberItem.detectedCardsDidChange(detectedCards: cardTypes)
             self.cardComponentDelegate?.didChangeCardType(cardTypes, component: self)
```

Comment:
```
self.numberItem.didChange(detectedCards: cardTypes)
```
would be more swifty IMO.

also 

```
self.cardComponentDelegate?.didChange(cardTypes, component: self)
```

but I guess its too late for this one, right?

Reply:
renamed